### PR TITLE
feat: support soft launch of artwork lists for Artsy employe

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [] 

#### NOT Artsy employee
| FF Disabled | FF Enabled |
| ------------- | ------------- |
|  |  |

#### Artsy employee
| FF Disabled | FF Enabled |
| ------------- | ------------- |
| Simulator Screen Shot - iPhone SE (3rd generation) - 2023-05-26 at 18 37 00 | Simulator Screen Shot - iPhone SE (3rd generation) - 2023-05-26 at 18 37 13 |

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- support soft launch of artwork lists for Artsy employe - dimatretyak

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Components/ArtworkLists/useCheckIfArtworkListsEnabled.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { GlobalStore } from "app/store/GlobalStore"
+import { useFeatureFlag } from "app/utils/hooks/useFeatureFlag"
+
+export const useCheckIfArtworkListsEnabled = () => {
+  const isArtsyEmployee = GlobalStore.useAppState((store) => store.auth.userHasArtsyEmail)
+  const isArtworkListsEnabled = useFeatureFlag("AREnableArtworkLists")
+
+  return isArtsyEmployee && isArtworkListsEnabled
```
Comment (anandaroop) [inline src/app/Components/ArtworkLists/useCheckIfArtworkListsEnabled.ts:8]: 👍🏽

**Review by anandaroop — APPROVED** [review]
Looks good!

#minor question — is the top right screenshot in the description correct? It shows `Feature Flag: Disabled`, which doesn't match the conditions?

**Review by dblandin — APPROVED** [review]
Awesome. Thanks for quick turnaround on this 👍🏼

File: src/app/Components/ArtworkLists/useCheckIfArtworkListsEnabled.ts (hunk 1 above)
Comment (dimatretyak) [reply src/app/Components/ArtworkLists/useCheckIfArtworkListsEnabled.ts:8]: > is the top right screenshot in the description correct? It shows Feature Flag: Disabled, which doesn't match the conditions?

yep, it's correct (for case when de decide to set feature flag to `false` on Echo side)

**Pull request merged**